src/functions/Invoke-AzOpsPush.ps1

```diff
@@ -164,6 +177,17 @@
             }
             else {
                 Write-PSFMessage -Level Verbose @common -String 'Invoke-AzOpsPush.Resolve.ParameterNotFound' -StringValues $FilePath, $parameterPath
+                if ((Get-PSFConfigValue -FullName 'AzOps.Core.AllowMultipleTemplateParameterFiles') -eq $true) {
+                    # Check for template parameters without defaultValue
+                    $defaultValueContent = Get-Content $FilePath
+                    $missingDefaultParam = $defaultValueContent | jq '.parameters | with_entries(select(.value.defaultValue == null))' | ConvertFrom-Json -AsHashtable
+                    if ($missingDefaultParam.Count -ge 1) {
+                        # Skip template deployment when template parameters without defaultValue are found and no parameter file identified
+                        $missingString = foreach ($item in $missingDefaultParam.Keys.GetEnumerator()) {"$item,"}
+                        Write-PSFMessage -Level Verbose -String 'Invoke-AzOpsPush.Resolve.NotFoundParamFileDefaultValue' -StringValues $FilePath, ($missingString | Out-String -NoNewline)
+                        continue
```

Comment:
As far as I can see, changing the template file without also changing the parameter files will cause the template to not be redeployed if it has parameters without default values. Will this not cause the "desired state" in the AzOps repository to become inconsistent with the actual, deployed state? After all, the template was originally deployed for each corresponding parameter file, so looking in the AzOps repository I would expect this to always be true.